Added bounds checking for percentage in SimpleProportionalStrategy

ClifHouck opened the pull request:
With an exception and tests.

aweeks commented on arsenal/strategy/simple_proportional_strategy.py:
```diff
@@ -111,8 +112,18 @@ def how_many_nodes_should_cache(nodes, percentage_to_cache):
 class SimpleProportionalStrategy(object):
     def __init__(self, percentage_to_cache=25):
+        # Clamp the percentage to reasonable values.
+        if percentage_to_cache < 0 or percentage_to_cache > 100:
```
We should use p ∈ [0, 1] to be consistent with style used in other stats libraries, and science.

aweeks replied:
I got a question from @pcsforeducation about this.  To be clear, I'm saying we should use probabilities, which are between 0 and 1inclusive, and avoid the informal percent (0 to 100 inclusive) notation.

ClifHouck replied:
+1 for science, I will update it accordingly.

pcsforeducation commented:
+1 LGTM

aweeks commented:
Otherwise, LGTM

ClifHouck commented:
Closing this as during addressing comments of this PR I discovered other issues that I've since addressed in a new PR, along with issues raised in the review of this PR.
